### ISSUE TITLE
Connection issue to public-repo

### ISSUE THREAD

**Issue by duncdrum** [eXist-db/existdb-packagemanager#6]
PackageManager on localhost shows installed packages but 0 available packages. 

It should show everything from exist-db.org's public repo. There are no connection issues to the public repo as the old package-manager shows the correct number of available apps and updates

There are no errors reported in the exist.log, but console on FF shows 1 error and one deprecation warning
```
[Deprecation] The “vaadin-valo-styles” (previously named “vaadin-valo-theme”) package is deprecated. Please migrate to the “vaadin-lumo-styles” package. See  https://github.com/vaadin/vaadin-valo-styles/blob/master/README.md  for more information. index.html:236:1


TypeError: h is not iterable index.html:198:5231

```

Safari is a bit more expressive with 3 errors.
```
[Error] Failed to load resource: the server responded with a status of 400 (Bad Request) (remote, line 0)
[Error] Unhandled Promise Rejection: TypeError: undefined is not a function
	(anonymous function) (index.html:198:83384)
	promiseReactionJob
[Error] Failed to load resource: the server responded with a status of 404 (Document /db/apps/packagemanager/webcomponents-lite.js.map not found) (webcomponents-lite.js.map, line 0)
```
![screen shot 2018-04-05 at 11 06 36](https://user-images.githubusercontent.com/6205362/38356915-6e83c78a-38c1-11e8-916f-b3ede3757490.png)

Tested on macOS with latest FF/Chrome/Safari.
exist `4.0.0`
PackageManager `4.5.0`
PackageService `1.3.0`

**Comment by JoernT** [comment]
@duncdrum the TypeError is a result of the connection problem which i assume must be somehow a local problem - at least it works fine for me in all browsers. However certainly there could be a better error detection for such cases.

**Comment by JoernT** [comment]
@duncdrum another tip: you can test if your packageservice works by accessing this url http://localhost:8080/exist/apps/packageservice/packages/remote in the browser. That should return you a page with repo-app elements (if you look at the pagesource)

**Comment by duncdrum** [comment]
@JoernT that was a good one, progress
```xml
<exception>
<path>
/db/apps/packageservice/modules/remote-packages.xql
</path>
<message>
exerr:ERROR XPTY0004: The actual cardinality for parameter data($pkg/child::{}icon) does not match the cardinality declared in the function's signature: zero or one. Expected cardinality: data($pkg/child::{}icon), got zero or one. [at line 27, column 71, source: /db/apps/packageservice/modules/remote-packages.xql]
</message>
</exception>
```

**Comment by JoernT** [comment]
@duncdrum still i can't reproduce that - works fine for me

**Comment by JoernT** [comment]
same versions and all

**Comment by duncdrum** [comment]
Checked with a dedicated docker image and with add-on free safari tech preview. (also ff and chrome). Same errors in the console for the new, while old package-manager connects just fine.
Not sure what's going on here.

**Comment by JoernT** [comment]
@duncdrum at least it seems it's more a problem of the service on your machine right? So it's not actually a PackageManager problem but a packageservice problem. I'd like to have tests but still being unexperienced on how to approach that.

**Comment by duncdrum** [comment]
@JoernT I experience this on two machines, and from within docker. So whatever my machines do differently it might actually affect more users. But you are right this should be a ticket for the packageservice repo, my bad. Just move this thing over. I currently don't understand whats going on well enough to come up with an idea for a test either.

**Comment by duncdrum** [comment]
Fixed by 1.3.1
